File: backend/app/schemas/knowledge.py
from datetime import datetime

from app.schemas.base import BaseSchema


class KnowledgeDocument(BaseSchema):
    id: str | None = None

    user_id: str

    filename: str

    original_filename: str

    content_type: str

    storage_path: str

    file_size: int

    page_count: int

    chunk_count: int = 0

    status: str = "uploaded"

    source: str = "upload"

    uploaded_at: datetime


class KnowledgeDocumentResponse(BaseSchema):
    """Public DTO returned to API consumers — excludes internal paths."""

    id: str

    user_id: str

    filename: str

    original_filename: str

    content_type: str

    file_size: int

    page_count: int

    chunk_count: int

    status: str

    source: str

    uploaded_at: datetime


class KnowledgeChunk(BaseSchema):
    id: str | None = None

    document_id: str

    user_id: str

    page_number: int

    chunk_index: int

    text: str

    created_at: datetime

File: frontend/src/types/knowledge.ts
export interface KnowledgeDocumentResponse {
    id: string;
    user_id: string;
    filename: string;
    original_filename: string;
    content_type: string;
    file_size: number;
    page_count: number;
    chunk_count: number;
    status: string;
    source: string;
    uploaded_at: string;
}

export interface RetrievalResult {
    chunk_id: string;
    document_id: string;
    page_number: number;
    chunk_index: number;
    text: string;
    score: number;
}

export interface DatasetResponse {
    name: string;
    display_name: string;
    description: string;
    source: string;
    is_ingested: boolean;
    document_count: number;
    chunk_count: number;
    ingested_at: string | null;
}

export interface DatasetIngestionResponse {
    ingested: string[];
    skipped: string[];
    failed: string[][];
    message: string;
}
